miniconf.py

```diff
@@ -62,29 +88,39 @@ async def command(self, path, value):
             value: The value to write to the path.
 
         Returns:
-            The received response to the command.
+            (code, msg) tuple as a response to the command. `code` is zero for success and `msg` is
+            a use-readable message indicating further information.
         """
         setting_topic = f'{self.prefix}/settings/{path}'
         response_topic = f'{self.prefix}/response/{path}'
-        if response_topic in self.inflight:
-            # TODO use correlation_data to distinguish clients and requests
-            raise NotImplementedError(
-                    'Only one in-flight message per topic is supported')
+
+        # Assign a unique identifier to this update request.
+        pid = self.request_id
+        self.request_id += 1
+        assert pid not in self.inflight, 'Invalid PID encountered'
+
+        correlation_data = json.dumps({
+            'id': self.uuid.hex,
+            'pid': pid,
+        }).encode('ascii')
 
         value = json.dumps(value)
-        logger.info('Sending %s to "%s"', value, setting_topic)
+        self.logger.info('Sending %s to "%s"', value, setting_topic)
         fut = asyncio.get_running_loop().create_future()
-        self.inflight[response_topic] = fut
+
+        self.inflight[pid] = fut
         self.client.publish(setting_topic, payload=value, qos=0, retain=True,
-                            response_topic=response_topic)
+                            response_topic=response_topic,
```

Comment:
I wonder if it might be better to dispense with the `uuid` in the correlation data and instead have one response topic per client (`{self.prefix}/response/{uuid}` without `path`). I.e. let the broker do the routing again.
Not critical here but maybe worth a simplification/improvement issue.

Reply:
Yes, this is imho the design intent. I've added correlation data precisely to implement it like this (choose topic based on UUID, and then use correlation data to disambiguate between multiple in-flight requests).